### PR TITLE
Shenli/cleanup function

### PR DESCRIPTION
**Pull request by shenli**
@siddontang I pass all the test but I am not sure about Distinct and AggAllOpt in function parser. 
PTAL

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -1798,30 +1811,74 @@ PrimaryExpression:
...
-		x := yylex.(*lexer)
-		f, ok := $1.(*expressions.Ident)
-		if !ok {
-			x.err("", "expected identifier or qualified identifier")
-			return 1
-		}
+		$$ = []expression.Expression{}
+	}
+	/* select count(*) from table */
+|	'*'
```
Comment (siddontang) [inline parser/parser.y]: only count support *, maybe we can write it below with count?

Comment (shenli) [inline parser/parser.y]: OK

File: parser/parser.y [hunk 2]
```diff
@@ -1798,30 +1811,74 @@ PrimaryExpression:
...
+		$$ = []expression.Expression{ expressions.Value{Val: expressions.TypeStar("*")} }
+	}
+|	ExpressionList
+
+FunctionCall:
+	'(' AggAllOpt FunctionCallArgList ')'
+	{
+		$$ = []interface{}{false, $3}
+	}
+|	'(' "DISTINCT" AggAllOpt ExpressionList')'
```
Comment (siddontang) [inline parser/parser.y]: only aggregate function needs distinct, should we remove it here?

but I don't know whether other functions use "all".

**Comment by siddontang** [comment]
Great work :+1: 

Rest LGTM.

**Comment by shenli** [comment]
PTAL

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**